mat64/solve.go

```diff
@@ -0,0 +1,87 @@
+// Copyright ©2015 The gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package mat64
+
+import (
+	"github.com/gonum/blas"
+	"github.com/gonum/lapack/lapack64"
+)
+
+// Solve solves a minimum-norm solution to a system of linear equations defined
+// by the matrices a and b. If a is singular or near-singular a Condition error
+// is returned. Please see the documentation for Condition for more information.
+//
+// The minimization problem solved depends on the input parameters.
+//  1. If m >= n and trans == false, find x such that || a*x - b||_2 is minimized.
```

Comment:
I agree I am also confused.

In solve, it should be a and b because they are actual inputs. In LU and QR it should be “A” since the matrix is implicitly represented. In all cases X should be capitalized since it is always implicit?

> On Aug 15, 2015, at 10:19 PM, Dan Kortschak <notifications@github.com> wrote:
> 
> In mat64/solve.go <https://github.com/gonum/matrix/pull/199#discussion_r37141796>:
> 
> > +// Use of this source code is governed by a BSD-style
> > +// license that can be found in the LICENSE file.
> > +
> > +package mat64
> > +
> > +import (
> > +	"github.com/gonum/blas"
> > +	"github.com/gonum/lapack/lapack64"
> > +)
> > +
> > +// Solve solves a minimum-norm solution to a system of linear equations defined
> > +// by the matrices a and b. If a is singular or near-singular a Condition error
> > +// is returned. Please see the documentation for Condition for more information.
> > +//
> > +// The minimization problem solved depends on the input parameters.
> > +//  1. If m >= n and trans == false, find x such that || a*x - b||_2 is minimized.
> I'm confused about the scheme that's being used to decide whether an entity is labelled with a capital or not. It seems to me that we should be using maths notation if we are describing the mathematical entity and the parameter literally if we are describing a paramter."x" is not a parameter, so it must be a mathematical matrix. At the very least the capitalisation should be consistently applied to each individual entity.
> —
> Reply to this email directly or view it on GitHub <https://github.com/gonum/matrix/pull/199/files#r37141796>.
>